src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultGeneratedDocumentPublisher.cs

```diff
@@ -85,6 +89,15 @@ public override void PublishCSharp(ProjectKey projectKey, string filePath, Sourc
 
         _projectSnapshotManagerDispatcher.AssertDispatcherThread();
 
+        // If our generated documents don't have unique file paths, then using project key information is problematic for the client.
+        // For example, when a document moves from the Misc Project to a real project, we will update it here, and each version would
+        // have a different project key. On the receiving end however, there is only one file path, therefore one version of the contents,
+        // so we must ensure we only have a single document to compute diffs from, or things get out of sync.
+        // if (!_languageServerFeatureOptions.IncludeProjectKeyInGeneratedFilePath)
```

Comment:
This line of code will be commented out in my other open PR, the feature flag doesn't exist in main yet.